paddle/gserver/layers/WithFunction.cpp

```diff
@@ -0,0 +1,24 @@
+#include "WithFunction.h"
+#include "paddle/function/Function.h"
+namespace paddle {
+
+void WithFunction::appendFunction(std::vector<function::KernelType> *functions,
+                                  const std::string &name,
+                                  const FuncConfig &config,
+                                  bool useGPU) {
+  std::shared_ptr<FunctionBase> func;
```

Comment:
I want to **replace** `FunctionBase` by plain `function` step by step.  The first step is to use `std::function` to **hide** it. 

The demo code of register function is here

```cpp
reg_cpu_function(cos, [](const BufferArgs& ins, const BufferArgs& outs, real scale){
    ...
});
```